### ISSUE TITLE
association \

### ISSUE THREAD

**Issue by sevu** [wesnoth/wesnoth#4439]

**Comment by GregoryLundberg** [comment]
What about Windows?

**Comment by Wedge009** [comment]
It's a bug? Or an enhancement? Generally I don't think they can be both.

**Comment by soliton-** [comment]
It's even documented to work the way it does.

**Comment by jostephd** [comment]
> It's even documented to work the way it does.

The man page says that `--load file loads the savegame file from the standard save game directory.` I can see how that means that `-l foo` when foo is a relative path would interpret foo relative to the saves directory. I cannot see how that means that `-l /bar` would just do string concatenation of the saves directory and `/bar` and try to open _that_. (On Windows the result wouldn't even be a valid pathname. On Linux it's just going to be a valid filename that doesn't exist.)

`-l /bar` should either treat `/bar` as an absolute path, or _possibly_ be an error, but it shouldn't pretend that /bar is a relative path. It isn't.

As I said in #2000, we really, _really_ should not invent our own rules for interpreting relative paths. We should just do what the rest of the world does. In this case, it is:

```
>>> os.path.join('/home/josteph/.local/share/wesnoth/1.14/saves', '/tmp/blargh') 
'/tmp/blargh'
```

(And to be clear, I agree that interpreting relative paths as relative to the saves dir makes sense in this specific case, notwithstanding #2000.)

**Comment by soliton-** [comment]
I don't agree that it makes sense to interprete relative paths as relative to anything but the current working directory. How is that helpful? It's certainly not helpful for a command line user that can't use tab completion for this. And sevu already explained that no other tool will expect that behaviour either.

Nevertheless it works as advertised.

Interpreting absolute paths as absolute is perfectly fine. No need to concatenate or whatever anything. Either way I don't think it's useful to argue about bug or not. If you want to call it a bug then do so but I agree with wedge that it can't be both.

I'm not sure if just fixing the absolute path behaviour is a big win. I suppose it's a step in the right direction but still confusing. Do we need to preserve backwards compatibility here? What's using this other than someone interactively from the command line?

If we do need backwards compatibility I'd rather use a new option.

**Comment by jostephd** [comment]
About absolute paths, I think we should just change it. I don't think we need compatibility for it either.

About relative paths, I suppose we could add a new option that interprets relative paths as relative to the current directory?

**Comment by soliton-** [comment]
I‘d either use a new option with the expected behavior for both cases or adjust both cases for the existing option. A mix is going to be confusing IMO.

**Comment by hexagonrecursion** [comment]
I think this issue should be renamed to "Register wesnoth as the app that opens wesnoth maps, saves and replays"

**Comment by AI0867** [comment]
> I think this issue should be renamed to "Register wesnoth as the app that opens wesnoth maps, saves and replays"

Doing that is what led sevu to find this problem, so that would be a separate issue. Another thing is that the extensions we use for those files are hardly unique, so we can't solely depend on those in most cases.
